### ISSUE TITLE
Can't specify hiera::hierarchy in hiera

### ISSUE THREAD

**Issue by westbywest** [hunner/puppet-hiera#92]
Having the default value for the "hierarchy" parameter specified as [], i.e. an empty array, prevents one from providing the value of this parameter via hiera.  You need to leave a default parameter value as undef for Puppet to do automatic parameter lookup.

https://docs.puppetlabs.com/hiera/3.0/puppet.html#automatic-parameter-lookup

**Comment by jyaworski** [comment]
Can you give an example of this failing?

**Comment by westbywest** [comment]
Yes.  Defining this array in hiera:
```
hiera::hierarchy:
  - "%{environment}/%{calling_class}"
  - "%{environment}"
  - common
```
... and then invoking the hiera class with expectation of it picking up the "hierarchy" parameter automatically from hiera fails:
```
class { 'hiera': }
```
The class uses '[]' for the parameter value.  Specifying a default empty array in code overrides Puppet's automated parameter lookup from hiera, as that lookup only does native merging and not deep merging.

The work-around I have to do is wrap hiera.pp in a another class (i.e. a profile class) and do the hiera lookup of "hiera::hierarchy" explicitly.

```
$hierarchy_real = hiera("hiera::hierarchy", undef)
class { 'hiera':
  hierarchy => hierarchy_real,
}
```

**Comment by hunner** [comment]
Merged in #119
